embedding/csharp/Tizen.Flutter.Embedding/ElmFlutterView.cs

```diff
@@ -0,0 +1,193 @@
+
+        /// <summary>
+        /// The initial width of the view. Defaults to the parent width if the value is zero.
+        /// </summary>
+        private int InitialWidth { get; set; } = 0;
+
+        /// <summary>
+        /// The initial height of the view. Defaults to the parent height if the value is zero.
+        /// </summary>
+        private int InitialHeight { get; set; } = 0;
+
+        /// <summary>
+        /// Whether the view is running.
+        /// </summary>
+        public bool IsRunning => !View.IsInvalid;
+
+        /// <summary>
+        /// The current width of the view.
+        /// </summary>
+        public int Width
+        {
+            get
+            {
+                Debug.Assert(IsRunning);
+
+                return EvasObject.Geometry.Width;
+            }
+        }
+
+        /// <summary>
+        /// The current height of the view.
+        /// </summary>
+        public int Height
+        {
+            get
+            {
+                Debug.Assert(IsRunning);
+
+                return EvasObject.Geometry.Height;
+            }
+        }
+
+        public ElmFlutterView(EvasObject parent)
+        {
+            Parent = parent;
+        }
+
+        public ElmFlutterView(EvasObject parent, int initialWidth, int initialHeight)
+        {
+            Parent = parent;
+            InitialWidth = initialWidth;
+            InitialHeight = initialHeight;
+        }
+
+        /// <summary>
+        /// Starts running the view with the associated engine, creating if not set.
+        /// </summary>
+        public bool RunEngine()
+        {
+            if (Parent == null)
+            {
+                TizenLog.Error("The parent object is invalid.");
+                return false;
+            }
+
+            if (Engine == null)
+            {
+                Engine = new FlutterEngine();
+            }
+            if (!Engine.IsValid)
+            {
+                TizenLog.Error("Could not create a Flutter engine.");
+                return false;
+            }
+
+            var viewProperties = new FlutterDesktopViewProperties
+            {
+                width = InitialWidth,
+                height = InitialHeight,
+            };
+
+            View = FlutterDesktopViewCreateFromElmParent(ref viewProperties, Engine.Engine, Parent);
+            if (View.IsInvalid)
+            {
+                TizenLog.Error("Could not launch a Flutter view.");
+                return false;
+            }
+
+            EvasObject = new EvasObjectImpl(Parent, FlutterDesktopViewGetEvasObject(View));
+            if (!EvasObject.IsRealized)
+            {
+                TizenLog.Error("Could not get an Evas object.");
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Sets an engine associated with this view.
+        /// </summary>
+        public void SetEngine(FlutterEngine engine)
```

Comment:
Is this setter necessary? If this setter is used after `RunEngine()` is called, it may be a problem. If necessary, why don't we move to the constructor's parameter or RunEngine's optional parameter?

Reply:
This API was designed to be consistent with C++'s

https://github.com/flutter-tizen/flutter-tizen/blob/8d6f09c254575b3e66b7d070e78adaec9fa18ed3/embedding/cpp/elm_flutter_view.cc#L54-L56

and yes, we can change the API as you said. However, if we do so, the C++ API will need a change as well and transferring the ownership of `std::unique_ptr<FlutterEngine>` will become less explicit:

```cpp
// |engine| can be nullptr.
bool ElmFlutterView::RunEngine(std::unique_ptr<FlutterEngine> engine) {
  // If engine is nullptr, create one.
  ...
}
```

Is this fine? @bbrto21 How do you think?

Reply:
if the engine can be used as a parameter in any form, there is still a problem that the engine can run before set engine to view. for now I think, it seems better to deal with only the parameters for the dart project rather than using the engine as a parameter.